homeassistant/components/sensor/speedtest.py

```diff
@@ -143,9 +146,12 @@ def update(self, now):
 
         _LOGGER.info('Executing speedtest')
         try:
+            args = [sys.executable, speedtest_cli.__file__, '--simple']
+            if self._server_id:
```

Comment:
This will always be true since you copy it to a string above. Even when it was `None`, it will have become `'None'`, which will return `True`.